src/LightGBM.jl

```diff
@@ -39,7 +40,16 @@ end
 
 
 function __init__()
-    LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
+    if Sys.isapple()
```

Comment:
This conditional does not check the right thing. It will incorrectly enter lines 44-47 for x86 Apple systems and will incorrectly enter lines 49-50 for ARM chips such as the ones by Ampere Computing (https://amperecomputing.com/) or the well known Raspberry Pi. It will also not be ready for RISC-V chips that are coming in.

It's probably the safest bet to default always to BinaryBuilder and let that system handle all the different architectures that exist and will exist in the future. Ever since the slowing of Moore's law, "things will only get weirder from here", according to Chris Lattner. Luckily, BinaryBuilder is very future proof with their system.

Reply:
This condition works for both x86_64 and aarch64 on macOS (given that most of the installation issues were for M1 users) but it is correct it will not work for non-Apple ARM architectures as it will trigger the other condition which is the current status quo. Of course it can get weirder ;) I'll have a look into a suggestion of overriding the jll artifacts to allow for custom binaries as I wouldn't want to compromise on this as we do have some use cases for custom binaries.